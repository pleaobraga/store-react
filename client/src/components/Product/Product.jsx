import React from 'react'
import PropTypes from 'prop-types'
import ListItem from '../ListItem'
import './Product.scss'

export const Product = ({ id, name, quantity, price, currency, onChange }) => {
  return (
    <ListItem
      id={id}
      onChange={onChange}
      className="product"
      details={
        <>
          <p key={name} className="product__name">
            {name}
          </p>
          <p key={quantity} className="product__quantity">
            {quantity}
          </p>
          <p key={price} className="product__price">
            <span>{currency}</span> {price}
          </p>
        </>
      }
    />
  )
}

Product.propTypes = {
  id: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired,
  price: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired,
  onChange: PropTypes.func
}

export default React.memo(Product)
