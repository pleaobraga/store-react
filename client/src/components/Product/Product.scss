@import '../../styles/colors';

.product {
  .list-item__details {
    .product__name {
      width: 60%;
    }

    .product__quantity {
      width: 15%;
      text-align: right;
    }
    .product__price {
      width: 35%;
      text-align: right;
    }
  }
}
