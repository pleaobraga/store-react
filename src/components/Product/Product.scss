@import '../../styles/colors';

.product {
  .list-item__details {
    .product__name {
      width: 60%;
    }

    .product__quantity {
      width: 15%;
    }
    .product__price {
      width: 35%;
    }
  }
}
