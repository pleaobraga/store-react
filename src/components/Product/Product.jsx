import React from 'react'
import PropTypes from 'prop-types'
import ListItem from '../ListItem'
import './Product.scss'

export const Product = ({ name, quantity, price, currency }) => {
  return (
    <ListItem
      className="product"
      details={[
        <p key={name} className="product__name">
          {name}
        </p>,
        <p key={name} className="product__quantity">
          {quantity}
        </p>,
        <p key={price} className="product__price">
          <span>{currency}</span> {price}
        </p>
      ]}
      actions={['delete', 'edit']}
    />
  )
}

Product.propTypes = {
  name: PropTypes.string.isRequired,
  quantity: PropTypes.number.isRequired,
  price: PropTypes.string.isRequired,
  currency: PropTypes.string.isRequired
}

export default React.memo(Product)
