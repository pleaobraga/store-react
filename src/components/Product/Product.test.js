import React from 'react'
import { shallow } from 'enzyme'
import Product from './Product'

describe('Product', () => {
  const product = shallow(<Product title="Test" text="test" />)

  it('render proprely', () => {
    expect(product).toMatchSnapshot()
  })
})
