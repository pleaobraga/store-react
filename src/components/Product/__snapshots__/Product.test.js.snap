// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Product render proprely 1`] = `
<div
  className="product"
>
  <div
    className="product-details"
  >
    <p
      className="product__name"
    />
    <p
      className="product__quantity"
    />
    <p
      className="product__price"
    >
      <span />
       
    </p>
  </div>
  <div
    className="product-actions"
  >
    <i
      className="material-icons "
    >
      delete
    </i>
    <i
      className="material-icons "
    >
      edit
    </i>
  </div>
</div>
`;
